src/d3/hapticplot.d3.spec.ts

```diff
@@ -32,29 +41,39 @@ describe('VR Haptic Plot', () => {
   });
 
   it('places points for each datum in a one datum array', () => {
-    hapticplot.init(scene, [10]);
-    const expectedPosArray = [new Vector3(0, 10, -1)];
+    const dataArray = [10];
+    graphScale.domain([0, d3.max(dataArray) as number])  // max of dataset
+      .range([0, 0.5]);
+    hapticplot.init(scene, dataArray);
+    const expectedPosArray = Array.from(dataArray).map(
+      (datum: number, i) =>
+      new Vector3((0.5 / dataArray.length) * i, graphScale(datum) + 1, -1));
```

Comment:
I understand the impulse to generate the expected array from the input array, because it makes more clear the connection between the input and expected values, and also it means that changing something in one place will automatically update it in another. I've done this in the past.

But in tests, it's usually better to have explicitly constructed expected values. The more logic you have in your test case, the more likely there's an error with your test, and not with the code you're testing; after all, you're not testing your test.  In this case in particular there is an even bigger issue, which is that the code you're using in your tests is exactly the same as the code in your class that you are trying to test.  So if you made an error in your logic in one place, it's going to show up in the other as well.

(side note): we have (had)? this thing at Google called "Testing on the Toilet" which puts a little piece of paper inside each toilet stall with some tips on good testing practices.  This exact topic was covered in one of the issues, so this is a very common pattern, even among regular Googlers.